includes/class-wc-install.php

```diff
@@ -494,58 +494,58 @@ private static function get_schema() {
   KEY consumer_secret (consumer_secret)
 ) $collate;
 CREATE TABLE {$wpdb->prefix}woocommerce_attribute_taxonomies (
-  attribute_id bigint(20) NOT NULL auto_increment,
+  attribute_id BIGINT UNSIGNED NOT NULL auto_increment,
   attribute_name varchar(200) NOT NULL,
-  attribute_label longtext NULL,
-  attribute_type varchar(200) NOT NULL,
-  attribute_orderby varchar(200) NOT NULL,
+  attribute_label varchar(200) NULL,
+  attribute_type varchar(20) NOT NULL,
+  attribute_orderby varchar(20) NOT NULL,
   attribute_public int(1) NOT NULL DEFAULT 1,
   PRIMARY KEY  (attribute_id),
   KEY attribute_name (attribute_name($max_index_length))
```

Comment:
You can further optimise index size by reducing the prefix length from the max_index_length to something specific and appropriate.
For example, if you were to index last names in a school, you already get wonderful selectivity with 2 characters (85% unique), and 3 characters has 90% uniqueness for an average name distribution. This is sufficient, while vastly reducing the size of the index (shorter keys means more keys on a page, so fewer pages to retrieve to find what's needed - even if the server then then need to check a few rows for the exact value match, that's quicker).